qiskit/circuit/library/standard_gates/u3.py

```diff
@@ -27,7 +27,7 @@ class U3Gate(Gate):
 
     .. math::
         U3(\theta, \phi, \lambda) =
-            RZ(\phi - \pi/2) RX(\pi/2) RZ(\pi - \theta) RX(\pi/2) RZ(\lambda - \pi/2)
+            RZ(\phi + \pi) RX(\pi/2) RZ(\theta + \pi) RX(\pi/2) RZ(\lambda)
```

Comment:
It looks like this was using the zxz decomposition instead of zyz. For this I believe the device uses `R_z(\phi) R_x(-\pi/2) R_z(\theta) R_x(\pi/2) R_z(\lambda)`.

Reply:
Thanks for your message @ewinston , is this how U3 is implemented in the real backends then? It would be useful to write this somewhere in the docs - I've been trying to find this out for a while but struggling to get a reply. Alternatively @ajavadia said my version is consistent with Qiskit's decomposer in `quantum_info` so might it be better to use that?

Reply:
Is this referring to the `OneQubitEulerDecomposer`? It looks like transpiling to an IBM backend decomposes to u3 whereas what you have looks like the "U1X" basis in `OneQubitEulerDecomposer`. Perhaps I have the wrong reference?

Reply:
Yes, you have the right reference. I have 2 points:
1) I am happy replacing with `R_z(\phi) R_x(-\pi/2) R_z(\theta) R_x(\pi/2) R_z(\lambda)` as is done in ibmq_armonk but ordinary users aren't told if this is true in all backends.
2) Given IBM isn't too keen to share this info for the other backends, it also makes sense to just use the U1X basis `OneQubitEulerDecomposer` and not make any claims about IBMQ backend transpilation accuracy (as is the norm in Circuit Library docs).

I'll defer to your judgement which is better but as it stands the docs are mathematically inaccurate.

Reply:
Your second point suggests terra uses u1x but I'm not sure where this is?

Reply:
It's more that U1X is the only basis in `OneQubitEulerDecomposer` that decomposes into `rx` gates with fixed rotation, and virtual `rz` gates carrying the Euler angles, which is how IBM backends work, is it not? ([Efficient Z gates for QC](https://arxiv.org/pdf/1612.00858.pdf)). 

We could of course add a decomposer into `R_z(\phi) R_x(-\pi/2) R_z(\theta) R_x(\pi/2) R_z(\lambda)` as well into `quantum_info`, but I'm still not really convinced this is what IBM backends are parsing U3 into at the hardware level as some of the other things in `armonk.defaults().cmd_def` look suspect to me. Eg you give the decomposition of `u2` as `rz(\phi) ry(\pi/2) rz(\lambda)` but here `ry(\pi/2)` can only really be implemented as an `rx(\pi/2)` drive pulse with pi/2 phase offset, so what is really happening is `rz(\phi - \pi/2) rx(\pi/2) rz(\lambda + \pi)` anyway.

So, in the absence of hardware-level clarity, why not just add something consistent with what we already have?

Reply:
@ewinston  see new PR at #4842